### ISSUE TITLE
Crash When Getting A Headshot With The Hell Rifle

### ISSUE THREAD

**Issue by Nodens99** [pa1nki113r/Project_Brutality#1086]
When using the caustic variant of the hell rifle, headshots seem to crash the game. It locks up or freezes, requiring me to end gzdooms task in windows.

**Comment by JMartinez9820** [comment]
@Nodens99 Without a proper description, we don't know what this issue is. If there is no response within the next 24 hours, this will be closed.

**Comment by JMartinez9820** [comment]
Oh, and what monsters does this affect

**Comment by Nodens99** [comment]
> Oh, and what monsters does this affect

Seems to be Humanoid enemies, like zombie men. Could be more though.

**Comment by Nodens99** [comment]
> Oh, and what monsters does this affect

After some more play testing, it seems to possibly not be headshots but chain gunner enemies and their replacements that cause the crash.

**Comment by JMartinez9820** [comment]
@Nodens99 This should be possibly fixed in the Staging branch, try that out

**Comment by SchnozBott** [comment]
I think I have reproduced a variation of this crash bug, 100% reproducible the way I've been doing it.

It occurs when hitting chaingun guy variants with the rocket launcher, game hard locks during the explosion animation. The crash does not occur in the old JMartinez fork that I was given last year to nerf monster grenades. I updated to GZDoom 4.10.0, using the TNT.wad but I haven't tried it with other wads yet.
